pkg/controller/routes.go

```diff
@@ -216,18 +217,18 @@ func (c *Controller) setupRoutes() error {
 
 	// MCPCatalog
 	root.Type(&v1.MCPCatalog{}).HandlerFunc(mcpCatalog.Sync)
-	root.Type(&v1.MCPCatalog{}).HandlerFunc(mcpCatalog.DeleteUnauthorizedMCPServers)
-	root.Type(&v1.MCPCatalog{}).HandlerFunc(mcpCatalog.SetUpUserCatalogAuthorizations)
 
 	// MCPServerCatalogEntry
 	root.Type(&v1.MCPServerCatalogEntry{}).HandlerFunc(cleanup.Cleanup)
 
 	// MCPServer
 	root.Type(&v1.MCPServer{}).HandlerFunc(cleanup.Cleanup)
+	root.Type(&v1.MCPServer{}).HandlerFunc(mcpserver.DeleteOrphans)
+	root.Type(&v1.MCPServer{}).FinalizeFunc(v1.MCPServerFinalizer, credentialCleanup.RemoveMCPCredentialsForProject)
```

Comment:
The same finalizer can't be used on multiple handlers. What will happen is:
1. Before running the first finalizer function, we check that the finalizer is on the object. It is so run the first function.
2. The first function will run successfully.
3. The finalizer gets removed.
4. Before running the second function,  we check that the finalizer is on the object. It is not, so the second function doesn't get run.

Reply:
Oh, thanks. I will consolidate them into a single function.

Reply:
Fixed